typescript-generator/index.js

```diff
@@ -211,8 +211,8 @@ function buildGenerics (type, noDefault = false) {
 
   // generics can either be a value/instance_of or a named generics
   function buildGeneric (type) {
-    return typeof type === 'string'
-      ? (noDefault ? type : `${type} = unknown`)
+    return typeof type.namespace === 'string'
```

Comment:
Not sure why you are checking `namespace` here:
```suggestion
    return typeof type.name === 'string'
```

Reply:
Execution was failing because the `string` is changed to a `TypeName`. Since it's JS and not TS I'm not sure what other types can reach this function, so my thinking is that the `name` property was more likely to appear on other types than `namespace` that is very specific to `TypeName`.

Checking for the existence of `kind` would actually be more correct IHMO since this is what is used in `buildType`. So what about this:

```js
  function buildGeneric (type) {
    return typeof type.kind === 'string'
      ? buildType(type)
      : (noDefault ? type.name : `${type.name} = unknown`)
  }
```

Reply:
Go for it!

Reply:
Done. Please cross-check!